File: Low Level Design (LLD)/DesignPatterns/3. Structural/Iterator/Cpp/main.cpp
#include <iostream>
#include "./include/Library.h"

int main() {
    Library library;
    library.addBook(new Book("Design Patterns"));
    library.addBook(new Book("Effective C++"));
    library.addBook(new Book("Clean Code"));

    BookIterator* iterator = library.createIterator();

    while (iterator->hasNext()) {
        Book* book = iterator->next();
        std::cout << "Book: " << book->getTitle() << std::endl;
    }

    delete iterator;
    return 0;
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Iterator/Java/Main.java
public class Main {
    public static void main(String[] args) {
        Library library = new Library();
        library.addBook(new Book("Design Patterns"));
        library.addBook(new Book("Effective Java"));
        library.addBook(new Book("Clean Code"));

        BookIterator iterator = library.createIterator();

        while (iterator.hasNext()) {
            Book book = iterator.next();
            System.out.println("Book: " + book.getTitle());
        }
    }
}
